server/src/main/java/org/opensearch/identity/AuthenticationToken.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+/**
+ * Generic interface for all token formats to support to authenticate user, such as UserName/Password tokens, Access tokens, and more.
+ *
+ * @opensearch.experimental
+ */
+public interface AuthenticationToken {
+
```

Comment:
Should we comment, as a guideline, what formats should this support? or will that be taken into account once we finalize the implementation?

Reply:
I was loosely basing this on [AuthenticationToken.java](https://github.com/apache/shiro/blob/e1df06bd93ac7e843600a31aec735e7dce42df7d/core/src/main/java/org/apache/shiro/authc/AuthenticationToken.java), but the aspect of having a principal I think we might want to consider before we jump into it